src/benchmarks/micro/libraries/System.IO/StringReaderReadToEndTests.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.IO.Tests;
+
+[BenchmarkCategory(Categories.Libraries)]
+public class StringReaderReadToEndTests : TextReaderReadLineTests
+{
+    private const int InvocationsPerIteration = 400_000;
+
+    [GlobalSetup]
+    public void GlobalSetup() => _text = GenerateLinesText(LineLengthRange, 48 * 1024 * 1024);
+
+    [Benchmark(OperationsPerInvoke = InvocationsPerIteration)]
+    public void ReadLine()
+    {
+        for (int i = 0; i < InvocationsPerIteration; i++)
```

Comment:
please remove this loop, BenchmarkDotNet does it for you: https://github.com/dotnet/performance/blob/d7dac8a7ca12a28d099192f8a901cf8e30361384/docs/microbenchmark-design-guidelines.md#loops

Reply:
I was going by this:

"You also need to make sure that the benchmark itself performs enough of computations for a single invocation to run longer than 100ms. If you don't, your benchmark will be entirely invalid."